pkg/test/framework/components/echo/instances.go

```diff
@@ -213,3 +177,91 @@ func (i Instances) Contains(instances ...Instance) bool {
 	})
 	return len(matches) > 0
 }
+
+// Deployments is a set of Instances that share the same FQDN.
```

Comment:
Hey Steven, I'm still struggling a bit with understanding the distinction of `Deployments` vs `Instances`. Mostly my problem is that every time I see `Deployments`, I have to remember what it means.

Here is my understanding, please correct me if I'm wrong:

- A single `Instance` is equal to a single Kubernetes `Deployment`. It may be composed of multiple workloads either because of initial scaling configuration or because of rolling restarts.
- `Instances` are just a list of `Instance` (i.e. Deployment). The list can contain a mix and match of deployments for different services (i.e. FQDN).
- `Deployments` is a list of `Instances` where each entry contains only instances for a particular service (i.e. FQDN).

Assuming that's all correct, would a better name for `Deployments` be `Services` or `Fleets` (The term `fleet` is often used to refer to all of the workloads for a given service across all data centers)? I wonder if slightly changing the API might be helpful as well:

```golang
type Fleet interface {
    FQDN() string
    Instances() Instances
}

type Fleets []Fleet
```
 
Does that make sense? Is it an improvement in your mind?

Reply:
I just realized my comment here is wrong. `Deployments` is a set of `Instances` where each `Instances` shares the FQDN. This is a bit confusing.. 

I think your suggestion makes sense. Personally I like the name `Service` since in multi-cluster we like to think of a Service as spanning many clusters, made up of many deployments. 

The `Service` interface would probably be implemented by `[]Instance` and only cast when appropriate.

Reply:
I did the rename, I don't see any value for a `Service` type at the moment though... maybe it could have some convenience methods so we can avoid doing `[0]` to get the service name or check other values that should be common across a Service. 

```go
type Service interface {
  // Valid returns true if all instances have the same FQDN
  Valid() bool

  Name() string
  FQDN() string
}
```